db/migrate/20180104152128_correct_co2_emission_sliders.rb

```diff
@@ -0,0 +1,34 @@
+class CorrectCo2EmissionSliders < ActiveRecord::Migration
+  KG_PER_MJ_IN_G_PER_KWH = 3600.0
+
+  def change
+    scenarios = Scenario.where(
+      '(protected = ? OR created_at >= ?) AND source != ? AND title != ?',
+      true, 1.month.ago, 'Mechanical Turk', 'test'
+    ).order(:area_code)
+
+    count = 0
+    puts "Need to migrate: #{ scenarios.count } scenarios"
+
+    scenarios.find_each do |scenario|
+      next unless scenario.valid?
+
+      if count % 50 == 0
+        puts "#{ Time.now } | #{ count } -> #{ scenario.id }"
+      end
+
+      gql = scenario.gql
```

Comment:
@antw is there anything you can do that can make this migration any quicker? 😓

Reply:
> @antw is there anything that we can do that can make this migration any quicker? 😓

What is the purpose of this migration?

I understand the *what* (set CO2 emissions of imported electricity), but I don't get the *why*. As far as I know, in the model currently this will be zero; why do we want to update existing scenarios to have a non-zero value?

---

As for making it quicker, I think my solution would be to replace the exact calculation of CO2 (which requires that you calculate the scenario using `scenario.gql`) with an approximation. For example, instead take the CO2 emissions from a default scenario and use that instead of calculating it for every saved scenario. If it's a requirement for every scenario to have the exactly-correct value, that might not be an acceptable solution.

Reply:
@MartLubben actually wants to hold this off, since there's a bit of discussion about: 

> What is the purpose of this migration?

^ that exactly.

So I'm going to close this for now. If it's needed, I'll push the branch again and add an explanation about the what and why.